python/ray/data/_internal/batcher.py

```diff
@@ -73,6 +74,19 @@ def add(self, block: Block):
             block: Block to add to the block buffer.
         """
         if BlockAccessor.for_block(block).num_rows() > 0:
+            import pyarrow
+
+            # pyarrow.Table.slice is slow when the table has many chunks
+            # so we combine chunks into a single one to make slice faster
+            # with the cost of an extra copy.
+            # See https://github.com/ray-project/ray/issues/31108 for more details.
```

Comment:
Can we file an upstream bug with Arrow here? It would nice to be able to do this in the future, since presumably this significantly increases our peak memory usage.

Reply:
I reached out to them via the mailing list. I can also file a GH issue.